File: InvertBT.py
"""
Invert a binary tree.

     4
   /   \
  2     7
 / \   / \
1   3 6   9
to
     4
   /   \
  7     2
 / \   / \
9   6 3   1
"""

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


def invertTree(root):
    """
    :type root: TreeNode
    :rtype: TreeNode
    """
    if root!=None:
        helper(root)
    return root
def helper(root):
    temp = root.left
    root.left = root.right
    root.right = temp
    if root.left!=None:
        helper(root.left)
    if root.right!=None:
        helper(root.right)

File: InvertBT.java
class TreeNode{
    int val;
    TreeNode left;
    TreeNode = right;
    public TreeNode(int value){
        val = value;
    }
}
public class InvertBT{
    public void invertBT(TreeNode root){
        if(root==null)
            return;
        invertBT(root.left);
        invertBT(root.right);
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
    }

    public static void main(String[] args){
        InvertBT sol = new InvertBT();
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.right.left = new TreeNode(5);
        sol.invertBT(root);
        System.out.println(root.left.val);
    }
}